service/telemetry.go

```diff
@@ -213,15 +221,39 @@ func (tel *telemetryInitializer) initOpenCensus(cfg telemetry.Config, telAttrs m
 	return pe, nil
 }
 
-func (tel *telemetryInitializer) initOpenTelemetry() (http.Handler, error) {
-	// Initialize the ocRegistry, still used by the process metrics.
-	tel.ocRegistry = ocmetric.NewRegistry()
+func (tel *telemetryInitializer) initOpenTelemetry(attrs map[string]string, logger *zap.Logger) (http.Handler, error) {
+	logger.Info("Setting up telemetry with otel-go")
+
+	resAttrs := []attribute.KeyValue{
+		attribute.String(semconv.AttributeServiceName, "io.opentelemetry.collector"),
+	}
+
+	for k, v := range attrs {
+		resAttrs = append(resAttrs, attribute.String(k, v))
+	}
+
+	res, err := resource.New(context.TODO(), resource.WithAttributes(resAttrs...))
+	if err != nil {
+		return nil, fmt.Errorf("error creating otlp resources: %w", err)
+	}
 
 	exporter := otelprom.New()
-	tel.mp = sdkmetric.NewMeterProvider(sdkmetric.WithReader(exporter))
+	tel.mp = sdkmetric.NewMeterProvider(
+		sdkmetric.WithResource(res),
+		sdkmetric.WithReader(exporter),
+	)
+
+	global.SetMeterProvider(tel.mp)
 
 	registry := prometheus.NewRegistry()
-	if err := registry.Register(exporter.Collector); err != nil {
+
+	promLabels := make(prometheus.Labels)
+	for k, v := range attrs {
+		promLabels[sanitizePrometheusKey(k)] = v
+	}
+
+	wrappedRegisterer := prometheus.WrapRegistererWithPrefix("otelcol_", prometheus.WrapRegistererWith(promLabels, registry))
+	if err := wrappedRegisterer.Register(exporter.Collector); err != nil {
```

Comment:
Let's split this change in a separate PR.

Reply:
Created https://github.com/open-telemetry/opentelemetry-collector/pull/6223 for this.